### PR TITLE
[libovsdb] Modify matchers to take into account optional field values

### PR DESCRIPTION
**Pull request by alexanderConstantinescu**
When verifying if https://github.com/ovn-org/libovsdb/pull/220 works, I realized our testing framework needs to be modified to be able to handle optional field values (referenced by pointers). Explicitly for optional field values representing UUIDs (see: https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/nbdb/logical_router_static_route.go#L18) we need to compare the referenced pointer value and see if they match. 

I always get a bit nervous when using `reflect` and was a bit afraid that my changes in `matchers.go` will panic for non-string pointer values, but according to the doc for `String()`:

> func (reflect.Value).String() string
> (reflect.Value).String on pkg.go.dev

> String returns the string v's underlying value, as a string. String is a special case because of Go's String method convention. Unlike the other getters, it does not panic if v's Kind is not String. Instead, it returns a string of the form "<T value>" where T is v's type. The fmt package treats Values specially. It does not call their String method implicitly but instead prints the concrete values they hold.

So maybe we're fine? 

/assign @jcaamano @flavio-fernandes  

**- What this PR does and why is it needed**

**- Special notes for reviewers**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: go-controller/pkg/testing/libovsdb/matchers.go [hunk 1]
```diff
@@ -143,6 +143,14 @@ func testDataEqual(x, y TestData, ignoreUUIDs bool) bool {
 		f1 := v1.Field(i)
 		f2 := v2.Field(i)
 		switch f1.Kind() {
+		case reflect.Ptr:
```
Comment (jcaamano) [inline go-controller/pkg/testing/libovsdb/matchers.go:146]: Should we check the underlying type is a String for clarity and efficiency?

Comment (jcaamano) [reply go-controller/pkg/testing/libovsdb/matchers.go:146]: So `f1.Elem().Kind() == reflect.String && ingnoreUUIDs` ...

**Review by flavio-fernandes — COMMENTED** [review]
/lgtm

**Pull request merged**